### PR TITLE
Pin portfinder@1.0.21. Fix #1606.

### LINKED ISSUE #1606: firebase functions:shell - Port 5000 is not open, could not start functions emulator
### [REQUIRED] Environment info

**firebase-tools: 7.2.2**

**Platform:**
macOS
### [REQUIRED] Test case
`firebase functions:shell`

### [REQUIRED] Steps to reproduce
1. `$ npm install -g firebase-tools`
2. `$ npm install firebase-functions@latest firebase-admin@latest --save`
3. `firebase functions:shell`
### [REQUIRED] Expected behavior
firebase shell runs as advertised

### [REQUIRED] Actual behavior
CLI Output: `Port 5000 is not open, could not start functions emulator.`
**firebase-debug.log**
```
[debug] [2019-08-18T03:02:45.322Z] ----------------------------------------------------------------------
[debug] [2019-08-18T03:02:45.324Z] Command:       /usr/local/bin/node /usr/local/bin/firebase functions:shell
[debug] [2019-08-18T03:02:45.324Z] CLI Version:   7.2.2
[debug] [2019-08-18T03:02:45.324Z] Platform:      darwin
[debug] [2019-08-18T03:02:45.324Z] Node Version:  v10.15.3
[debug] [2019-08-18T03:02:45.330Z] Time:          Sat Aug 17 2019 23:02:45 GMT-0400 (Eastern Daylight Time)
[debug] [2019-08-18T03:02:45.330Z] ----------------------------------------------------------------------
[debug] 
[2019-08-18T03:04:19.141Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[2019-08-18T03:04:19.141Z] > authorizing via signed-in user
[2019-08-18T03:04:19.142Z] [iam] checking project ddme-329f1 for permissions ["firebase.projects.get"]
[2019-08-18T03:04:19.143Z] >>> HTTP REQUEST POST https://cloudresourcemanager.googleapis.com/v1/projects/ddme-329f1:testIamPermissions
 permissions=[firebase.projects.get]
[2019-08-18T03:04:24.132Z] <<< HTTP RESPONSE 200 content-type=application/json; charset=UTF-8, vary=X-Origin, Referer, Origin,Accept-Encoding, date=Sun, 18 Aug 2019 03:04:23 GMT, server=ESF, cache-control=private, x-xss-protection=0, x-frame-options=SAMEORIGIN, x-content-type-options=nosniff, server-timing=gfet4t7; dur=111, alt-svc=quic=":443"; ma=2592000; v="46,43,39", accept-ranges=none, transfer-encoding=chunked

Error: Port 5000 is not open, could not start functions emulator.
```

### PR DESCRIPTION
**Pull request by yuchenshi**
See #1606.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -92,7 +92,7 @@
     "minimatch": "^3.0.4",
     "open": "^6.3.0",
     "ora": "^3.4.0",
-    "portfinder": "^1.0.13",
+    "portfinder": "1.0.21",
```
Comment (mbleigh) [inline package.json]: Why not `^1.0.21`?

Comment (yuchenshi) [reply package.json]: Because 1.0.22 introduced a breaking change https://github.com/http-party/node-portfinder/issues/85

**Comment by yuchenshi** [comment]
Once all comments are resolved please feel free to merge and release without me :)

File: package.json (hunk 1 above)
Comment (yuchenshi) [reply package.json]: Sorry, wrong link. https://github.com/http-party/node-portfinder/issues/89 describes better what we're experiencing.

Comment (yuchenshi) [reply package.json]: portfinder has released a fix. I've updated the code to read `^1.0.23`. This PR is no longer urgent. https://www.npmjs.com/package/portfinder/v/1.0.23

**Pull request merged**